collector/generic_collector.go

```diff
@@ -57,6 +62,68 @@ func (collector *GenericCollector) Name() string {
 
 //Returns collected metrics and the next collection time of the collector
 func (collector *GenericCollector) Collect() (time.Time, []v2.Metric, error) {
-	//TO BE IMPLEMENTED
-	return time.Now(), nil, nil
+	minNextColTime := collector.configFile.MetricsConfig[0].PollingFrequency
+	for _, metricConfig := range collector.configFile.MetricsConfig {
+		if metricConfig.PollingFrequency < minNextColTime {
+			minNextColTime = metricConfig.PollingFrequency
+		}
+	}
+	currentTime := time.Now()
+	nextCollectionTime := currentTime.Add(time.Duration(minNextColTime * time.Second))
+
+	uri := collector.configFile.Endpoint
+	response, err := http.Get(uri)
+	if err != nil {
+		return nextCollectionTime, nil, err
+	}
+
+	defer response.Body.Close()
+
+	pageContent, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nextCollectionTime, nil, err
+	}
+
+	metrics := make([]v2.Metric, len(collector.configFile.MetricsConfig))
+	var errorSlice []error
+
+	for ind, metricConfig := range collector.configFile.MetricsConfig {
+		regex, err := regexp.Compile(metricConfig.Regex)
+		if err != nil {
+			return nextCollectionTime, nil, err
+		}
+
+		matchString := regex.FindStringSubmatch(string(pageContent))
+		if matchString != nil {
+			if strings.Contains(matchString[1], ".") {
```

Comment:
My concern was the way in which the metric would be expressed : int, integer, int32, number ... 
But yes, I think this is a better approach. We can use enum for metric units if necessary